src/Cms/ModelWithContent.php

```diff
@@ -443,7 +444,10 @@ public function kirby()
 	 */
 	public function lock()
 	{
-		$dir = $this->contentFileDirectory();
+		$dir = $this->root();
+		if ($this::CLASS_ALIAS === 'file') {
+			$dir = dirname($dir);
+		}
```

Comment:
This whole method will be gone in v4 (or at least deprecated) because the old locking system will no longer be needed. So it's only a temporary change to keep the system working after merging this PR.

Reply:
Ok